{{ cookiecutter.project_slug }}/src/app/settings.py

```diff
@@ -59,6 +59,9 @@
     "corsheaders.middleware.CorsMiddleware",
     "whitenoise.middleware.WhiteNoiseMiddleware",
     "django.contrib.sessions.middleware.SessionMiddleware",
+    {%- if cookiecutter.enable_health_check == "y" %}
+    "helpers.health_check_middleware.HealthCheckMiddleware",
```

Comment:
I personally would move the middleware files to `app/` instead.

Reply:
I agree. I will move to the app folder with filename `middleware.py`, what do you think about that?

Reply:
Yes, but call it `middlewares.py`, so we can use this file for any other middleware. 
Try to abstract the middleware base, so it's easy for anyone to implement. Something like this:


```python
from abc import ABC, abstractmethod

class AppMiddleware(ABC):
    def __init__(self, get_response):
        self.get_response = get_response
    
    @abstractmethod
    def __call__(self, request):
        raise NotImplementedError('Subclass must overwrite __call__(self, request)') 
```

Then.. 


```python
class HealthCheckMiddleware(AppMiddleware):
    def __call__(self, request):
        if request.path.replace("/", "") == "health_check":
            request.META["HTTP_HOST"] = "localhost"
            if settings.ALLOWED_HOSTS and not settings.DEBUG:
                request.META["HTTP_HOST"] = settings.ALLOWED_HOSTS[0]
        return self.get_response(request)
````

Something into this line.

Reply:
Oh, make sure you actually leave the generated `middlewares.py` file, and simply Jinja if the `HealthCheckMiddleware`, so the middleware base class will stay.

Reply:
@natamvo @phsetti @rodolfolottin care to comment on the following? ^

Reply:
Nice!! I'll work on it.